snapcraft/internal/mangling.py

```diff
@@ -113,25 +113,25 @@ def handle_glibc_mismatch(*, elf_files: FrozenSet[elf.ElfFile],
                    'base. These are the GLIBC versions required by '
                    'the primed files that do not match and will be '
                    'patched:\n {}\n'
-                   'To work around this, the newer libc will be '
-                   'migrated into the snap, and these files will be '
-                   'patched to use it.'.format('\n'.join(formatted_list)))
+                   'For this to work, `libc6` will need to be explicitly '
+                   'added to `stage-packages` for this part. These files '
+                   'will be patched to use it.'.format(
+                       '\n'.join(formatted_list)))
```

Comment:
Hmm... this is a warning, but it's phrased like an error. Should it be an error instead? If not, and libc is actually staged, is there still a reason to show this warning?

Reply:
this is left over, the caller of this method resolves that doubt. I will add appropirate docstrings